selfservice/hook/web_hook.go

```diff
@@ -141,6 +174,11 @@ func (e *WebHook) execute(ctx context.Context, data *templateContext) error {
 		}
 
 		if resp.StatusCode >= http.StatusBadRequest {
+			if gjson.GetBytes(e.conf, "can_interrupt").Bool() {
+				if err := parseWebhookResponse(resp); err != nil {
```

Comment:
Partially out of habit :-) But this is an implementation assumption and caller might not be aware of this  and can cause issues when someone modifies `parseWebhookResponse()` and add a case where `nil` will be returned. Usually when function returns an `error` one might expect it to return `nil` in some cases (default Go contract) and keeping this logic would violate "least surprise" rule ;-)

Reply:
One more thing, we only read from `errChan` once in https://github.com/ory/kratos/pull/1585/files#diff-6dbaf3f7f3f6f427e72de26f22beb17b0c56379f702ae5be81516c38765f8d13R220, but are writing to it in lines 201 and 204. The second send (in 204) will still succeed, since the channel is buffered, but it is still cleaner to return after the first send in 201, since that is what will really happen, right?